app.rb

```diff
@@ -0,0 +1,147 @@
+require_relative 'rental'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'book'
+
+class App
+  def initialize()
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def app_options
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person id'
+    puts '7 - Exit'
+    puts "\n\n"
+  end
+
+  def run
+    user_response = 0
+    puts "Welcome to School Library App!\n\n"
+    while user_response != '7'
+      puts "Please choose an option by entering a number:\n\n"
+      app_options
+      print 'Enter Option [number]: '
+      user_response = gets.chomp
+      puts "\n\n"
+    end
+    puts "Thank you for using this app!\n\n" if user_response == '7'
+  end
+
+  def check_selection(response)
+    case response
+    when '1'
+      list_all_books
+    when '2'
+      list_all_people
+    when '3'
+      create_person
+    when '4'
+      create_book
+    when '5'
+      create_rental
+    when '6'
+      list_all_rentals_person_id
+    end
+  end
```

Comment:
@muneebulrehman Hi, I appreciate your suggestion. However, I have to disagree since the `app.run` method in the `main.rb` already handles all the input-related tasks. This is just a different approach that still meets the requirements for this project :100: